### ISSUE TITLE
Entity Summary: Produce and consume label - url pairs 

### ISSUE THREAD

**Issue by jvwong** [PathwayCommons/app-ui#1132]
## Description of new feature

The entity summary service provides external db, id data and the client maps this to urls. Entity summary service should provide label - url pairs for various external databases. Clients should consume and display this data.

## Motivation for new feature

Much simpler for client to display link data, rather than db, id pairs.

**Comment by IgorRodchenkov** [comment]
Shall we merge this with #1131 ?

**Comment by d2fong** [comment]
I don't think it should be merged yet.

**Comment by jvwong** [comment]
I'll keep this issue. First finish off #1131,  then here specifically address:

1. Rename `EntitySummary` model attribute from `xref` to `links`
2. In external services, `links` will be filled with key: namespace, value: URL for entity
3. UI code will format `links` as required

**Comment by jvwong** [comment]
Something like this:?

```js
	"TP53": {
		...
		"links": {
			"http://identifiers.org/hgnc.symbol/TP53": config.NS_HGNC_SYMBOL,
                        ...
		}
	}
```

**Comment by IgorRodchenkov** [comment]
Wait, I don't understand.. I did not mean use the uri (or any data value) as a key, no.
Why do we need a namespace if there is URI (link) already? 
Why not just 
`"TP53": {"links":["http://identifiers.org/hgnc.symbol/TP53",..]}`
If we want namespaces, let's return them separately, in another array or map...

**Comment by jvwong** [comment]
ok I see what you mean.

**Comment by d2fong** [comment]
If we just have an array of links how do we know which one is hgnc, which one is uniprot, etc.

**Comment by jvwong** [comment]
In this case, the client will have to find or parse the pattern `http://identifers.org/<namespace>/<localId>`. We have to search anyways to find the 'supported collections'

**Comment by IgorRodchenkov** [comment]
And for what use case do we want to know that as long as the links get resolved..
Ah, I see - "TP53" is a node label, and we want return all the links at once with each having label (perhaps ID) to display, etc..; 
then it should be smth. like or simpler:
```
"TP53": {
  ...
  "links": [
     {
        "uri" : "http://identifiers.org/hgnc.symbol/TP53",
       "namespace" : config.NS_HGNC_SYMBOL,
       "label" : "TP53"
     },...
  ]
}
```

**Comment by IgorRodchenkov** [comment]
What is 'oi'? (BTW, IDs are case sensitive)
This is another issue, if any. Let's not mess data format, model and whether those URI work or not and why ;) 

So,  it goes to `https://www.ebi.ac.uk/miriam/main/datatypes/MIR:00000323`
And, `http://identifiers.org/genecards/ABL1` should get me `http://www.genecards.org/cgi-bin/carddisp.pl?gene=ABL1` but gets `http://www.genecards.org/cgi-bin/carddisp.pl?gene=ABL1`  instead. 
I reported this issue to MIRIAM (feedback).

**Comment by jvwong** [comment]
Adding work from #1131 here as it is all self-contained 

>- [`external-services`](https://github.com/PathwayCommons/app-ui/tree/development/src/server/external-services) calls to NCBI, HGNC, UniProt uses these names
>- Entity Summary 
>    - [model](https://github.com/PathwayCommons/app-ui/blob/345fe33603b66d2fd8ac21d48ceeeec7a0fb3477/src/models/entity/summary.js#L5-L10)
>    - [UI box](https://github.com/PathwayCommons/app-ui/blob/345fe33603b66d2fd8ac21d48ceeeec7a0fb3477/src/client/features/search/entity-summary-box.js#L15-L32)
>    - [web service](https://github.com/PathwayCommons/app-ui/blob/345fe33603b66d2fd8ac21d48ceeeec7a0fb3477/src/server/routes/summary/entity/index.js#L6)

**Comment by IgorRodchenkov** [comment]
Ah, I made the point but forgot to finish editing that json data example: we should avoid using data values as keys in data transfer objects, i.e., shouldn’t be “TP53:” key; there is no need in this, and only complicates dev.

**Comment by jvwong** [comment]
> Ah, I made the point but forgot to finish editing that json data example: we should avoid using data values as keys in data transfer objects, i.e., shouldn’t be “TP53:” key; there is no need in this, and only complicates dev.

Are you saying put the input data value within the returned 'Summary' object? Otherwise, how am  I going to match the Summary object to an input gene (e.g. input='TP53 MDM2') ?

**Comment by IgorRodchenkov** [comment]
Correct, "label" : "TP53" is already inside the object in the result list (could be "input_name" or just "name" istead of "label" there)

**Comment by jvwong** [comment]
Now returning the following for an EntitySummary (`/api/summary/entity/search/?q=TP53`):

```js
[
	{
		"query": "TP53",
		"summary": {
			"namespace": "ncbigene",
			"displayName": "tumor protein p53",
			"localId": "7157",
			"description": "This gene encodes a tumor suppressor protein containing transcriptional activation, DNA binding, and oligomerization domains. The encoded protein responds to diverse cellular stresses to regulate expression of target genes, thereby inducing cell cycle arrest, apoptosis, senescence, DNA repair, or changes in metabolism. Mutations in this gene are associated with a variety of human cancers, including hereditary cancers such as Li-Fraumeni syndrome. Alternative splicing of this gene and the use of alternate promoters result in multiple transcript variants and isoforms. Additional isoforms have also been shown to result from the use of alternate translation initiation codons from identical transcript variants (PMIDs: 12032546, 20937277). [provided by RefSeq, Dec 2016]",
			"aliases": [
				"cellular tumor antigen p53",
				"antigen NY-CO-13",
				"mutant tumor protein 53",
				"p53 tumor suppressor",
				"phosphoprotein p53",
				"transformation-related protein 53",
				"tumor protein 53",
				"tumor supressor p53"
			],
			"aliasIds": [
				"BCC7",
				"BMFS5",
				"LFS1",
				"P53",
				"TRP53"
			],
			"xrefLinks": [
				{
					"namespace": "hgnc.symbol",
					"uri": "http://identifiers.org/hgnc.symbol/TP53"
				},
				{
					"namespace": "genecards",
					"uri": "http://identifiers.org/genecards/TP53"
				},
				{
					"namespace": "ncbigene",
					"uri": "http://identifiers.org/ncbigene/7157"
				},
				{
					"namespace": "uniprot",
					"uri": "http://identifiers.org/uniprot/P04637"
				}
			]
		}
	},
	...
]
```
